superset-frontend/plugins/plugin-chart-cartodiagram/test/util/transformPropsUtil.test.ts

```diff
@@ -17,124 +17,147 @@
  * under the License.
  */
 
-import { getChartTransformPropsRegistry } from '@superset-ui/core';
 import {
   groupByLocationTs,
   groupByLocation,
   getEchartConfigs,
   parseSelectedChart,
 } from '../../src/util/transformPropsUtil';
-import { barChartData, pieChartData } from '../testData';
+import {
+  timeseriesChartData,
+  nonTimeSeriesChartData,
+  groupedTimeseriesChartData,
+  geom1,
+  geom2,
+} from '../testData';
+
+describe('transformPropsUtil', () => {
+  describe('groupByLocationTs', () => {
+    it('returns empty result when no timestamps provided', () => {
+      const result = groupByLocationTs(nonTimeSeriesChartData);
+      expect(result).toEqual({});
+    });
 
-describe('groupByLocationTs', () => {
-  it('returns empty result when no timestamps provided', () => {
-    const result = groupByLocationTs(pieChartData);
-    expect(result).toEqual({});
-  });
-  it('returns correct structure on valid input', () => {
-    const result = groupByLocationTs(barChartData);
-    expect(false).toEqual(true);
-  });
-});
+    it('groups in the correct count of geometries', () => {
+      const result = groupByLocationTs(timeseriesChartData);
 
-describe('groupByLocation', () => {
-  const geometryColumn = 'geom';
-  const result = groupByLocation(pieChartData, geometryColumn);
+      const countGeometries = Object.keys(result).length;
+      expect(countGeometries).toEqual(2);
+    });
 
-  it('groups in the correct count of geometries', () => {
-    const countOfGeometries = Object.keys(result).length;
-    expect(countOfGeometries).toEqual(2);
+    it('removes the geometry from the data field for grouped data', () => {
+      const result = groupByLocationTs(groupedTimeseriesChartData);
+
+      expect(result).toEqual(
```

Comment:
could you write a short comment what is happening here?

Reply:
It's a way of checking only the relevant parts of a complex structure. This way, we do not have to provide all keys of an object, or all items of an array, but only those that we actually want to check.